test/migration/SushiMigrator.js

```diff
@@ -0,0 +1,166 @@
+const { fork_network } = require("../utils/network_fork");
+const impersonateAccount = require("../utils/impersonate_account");
+const chai = require("chai");
+const { assert, expect } = require("chai");
+const { solidity } = require("ethereum-waffle");
+const { olympus } = require("../utils/olympus");
+
+chai.use(solidity);
+
+const sushiRouter = "0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F";
+const uniRouter = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D";
+const olympusGovernor = "0x245cc372C84B3645Bf0Ffe6538620B04a217988B";
+
+describe("SushiMigrator", async () => {
+    let governor, treasury, ohmContract, SushiMigrator, sushiMigrator;
+
+    beforeEach(async () => {
+        await fork_network(14486473);
+        treasury = await ethers.getContractAt(
+            "OlympusTreasury",
+            "0x9A315BdF513367C0377FB36545857d12e85813Ef"
+        );
+        ohmContract = await ethers.getContractAt(
+            "contracts/interfaces/IERC20.sol:IERC20",
+            "0x64aa3364F17a4D01c6f1751Fd97C2BD3D7e7f1D5"
+        );
+
+        SushiMigrator = await ethers.getContractFactory("SushiMigrator");
+        sushiMigrator = await SushiMigrator.deploy(olympus.authority);
+
+        impersonateAccount(olympusGovernor);
+        governor = await ethers.getSigner(olympusGovernor);
+    });
+
+    it("Migrate Sushi OHM/DAI pair to Uniswap OHM/DAI pair", async () => {
+        const sushiOhmDaiLpAddress = "0x055475920a8c93CfFb64d039A8205F7AcC7722d3";
+        const uniOhmDaiLpAddress = "0x1b851374b8968393c11e8fb30c2842cfc4e986a5";
+        const amount = "253163960111968806387";
+
+        const uniswapLpContract = await ethers.getContractAt(
+            "contracts/interfaces/IERC20.sol:IERC20",
+            uniOhmDaiLpAddress
+        );
+        const daiContract = await ethers.getContractAt(
+            "contracts/interfaces/IERC20.sol:IERC20",
+            "0x6B175474E89094C44Da98b954EedeAC495271d0F"
+        );
+
+        await treasury.connect(governor).enable(3, sushiMigrator.address, sushiMigrator.address);
+
+        const ohmBalBeforeTx = await ohmContract.balanceOf(treasury.address);
+        const daiBalBeforeTx = await daiContract.balanceOf(treasury.address);
+        const lpBalBeforeTx = await uniswapLpContract.balanceOf(treasury.address);
+
+        await sushiMigrator
+            .connect(governor)
+            .executeTx(
+                sushiRouter,
+                uniRouter,
+                sushiOhmDaiLpAddress,
+                uniOhmDaiLpAddress,
+                amount,
+                990,
+                900
```

Comment:
It doesn't work with less than 10% slippage, so like 10% of underlying doesn't get added to LP?

Reply:
Yes, which is sent back to the treasury. The percentage may vary depending on the reserve tokens amount.